filament/src/details/Material.h

```diff
@@ -284,7 +290,10 @@ class FMaterial : public Material {
     bool mIsDefaultMaterial = false;
     bool mSpecularAntiAliasing = false;
 
-    FMaterialInstance mDefaultInstance;
+    // reserve some sopce to construct the default material instance
+    std::aligned_storage<sizeof(FMaterialInstance), alignof(FMaterialInstance)>::type mDefaultInstanceStorage;
+    static_assert(sizeof(mDefaultInstanceStorage) >= sizeof(mDefaultInstanceStorage));
```

Comment:
nit: may I ask for a little explanation on this assert?

Reply:
it's really not needed, I have it because if you forget the `::type` to the right of the expression, it compiles and you really don't get what you want (I did this before!), and that would catch it.  It's really just a way to be extra sure that I have enough space.